internal/xds/balancer/clusterresolver/configbuilder.go

```diff
@@ -139,24 +140,36 @@ func makeClusterImplOutlierDetectionChild(ciCfg *clusterimpl.LBConfig, odCfg out
 	return &odCfgRet
 }
 
-func buildClusterImplConfigForDNS(g *nameGenerator, endpoints []resolver.Endpoint, mechanism DiscoveryMechanism) (string, *clusterimpl.LBConfig, []resolver.Endpoint) {
-	// Endpoint picking policy for DNS is hardcoded to pick_first.
-	const childPolicy = "pick_first"
-	retEndpoints := make([]resolver.Endpoint, len(endpoints))
+func buildClusterImplConfigForDNS(g *nameGenerator, endpoints []resolver.Endpoint, mechanism DiscoveryMechanism, xdsLBPolicy *internalserviceconfig.BalancerConfig) (string, *clusterimpl.LBConfig, []resolver.Endpoint) {
 	pName := fmt.Sprintf("priority-%v", g.prefix)
-	for i, e := range endpoints {
-		retEndpoints[i] = hierarchy.SetInEndpoint(e, []string{pName})
+	if len(endpoints) == 0 {
+		return pName, &clusterimpl.LBConfig{
+			Cluster:               mechanism.Cluster,
+			TelemetryLabels:       mechanism.TelemetryLabels,
+			ChildPolicy:           xdsLBPolicy,
+			MaxConcurrentRequests: mechanism.MaxConcurrentRequests,
+			LoadReportingServer:   mechanism.LoadReportingServer,
+		}, []resolver.Endpoint{}
```

Comment:
Can this return a `nil` slice instead? It is not a good idea to have an API that differentiates between empty and `nil` slices. See: go/go-style/decisions#nil-slices